### PR TITLE
NAS-3371

### PR DESCRIPTION
**Pull request by hkad98**
NAS-3371 fix of missing id in data filter + added tests

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by petr-klemsinsky** [comment]
ok to test

**Comment by leosflodr** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/tests/catalog/test_catalog_workspace_content.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    path = _current_dir / "load" / "declarative_analytics_model" / "declarative_analytics_model_demo.yaml"
+    identifier = test_load_and_put_declarative_analytics_model.__name__
+    _set_up_workspace_ldm(sdk, test_config["workspace"], identifier)
+    analytics_model_e = sdk.catalog_workspace_content.get_declarative_analytics_model(test_config["workspace"])
+
+    try:
+        sdk.catalog_workspace_content.load_and_put_declarative_analytics_model(identifier, path)
+        analytics_model_o = sdk.catalog_workspace_content.get_declarative_analytics_model(identifier)
+        assert analytics_model_e == analytics_model_o
+        # assert analytics_model_e.to_api().to_dict() == analytics_model_o.to_api().to_dict()
```
Comment (jaceksan) [inline gooddata-sdk/tests/catalog/test_catalog_workspace_content.py:118]: Why is this commented out?

File: gooddata-sdk/tests/catalog/test_catalog_workspace_content.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        assert ldm_e == ldm_o
+        assert ldm_e.to_api().to_dict() == ldm_o.to_api().to_dict()
+    finally:
+        sdk.catalog_workspace.delete_workspace(identifier)
+
+
+@gd_vcr.use_cassette(str(_fixtures_dir / "demo_get_declarative_analytics_model.json"))
+def test_get_declarative_analytics_model(test_config):
+    sdk = GoodDataSdk.create(host_=test_config["host"], token_=test_config["token"])
+    path = _current_dir / "expected" / "declarative_analytics_model.json"
```
Comment (jaceksan) [inline gooddata-sdk/tests/catalog/test_catalog_workspace_content.py:159]: I would store these json files into "fixtures/workspace_content", not into "expected".

**Review by jaceksan — CHANGES_REQUESTED** [review]

File: gooddata-sdk/tests/catalog/test_catalog_workspace_content.py (hunk 1 above)
Comment (hkad98) [reply gooddata-sdk/tests/catalog/test_catalog_workspace_content.py:118]: Forgot. Thanks 👍

File: gooddata-sdk/tests/catalog/test_catalog_workspace_content.py (hunk 2 above)
Comment (hkad98) [reply gooddata-sdk/tests/catalog/test_catalog_workspace_content.py:159]: These files represent expected content. Do we really want to mix them with fixtures? There is also declarative_workspaces.json, I assume that this file should be part of the fixtures/workspaces right?

Comment (jaceksan) [reply gooddata-sdk/tests/catalog/test_catalog_workspace_content.py:159]: I get it, let's keep it like it is

**Review by jaceksan — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR82

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/74/ : SUCCESS in 55s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul-el7/74/ : SUCCESS in 2m 12s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/74/ : SUCCESS in 20s

**Pull request merged**